### DISCUSSION TITLE
PageInfo / Pagination

### DISCUSSION THREAD

**Discussion by ovu** [ExpediaGroup/graphql-kotlin#966] [Q&A] [answered]
Hi all,

I want to generate the pagination and edges in the schema. Is there any place in the documentation about how to use it?
I found this documentation about graphql https://graphql.org/learn/pagination/#pagination-and-edges, however, I could not find any sample in the documentation.

Thanks in advance!

**Comment by smyrick** [comment, marked as answer]
Hello @ovu!

GraphQL does not provide details on how to implement your schema. It just provides the tools to describe what inputs you accept as part of your API, the response you return, and allows clients to select what parts of the response they want. This means there is no paging in the language itself and would just be up to you to implement it however you want.

If you wanted to follow that pattern you have linked you could declare a function like so:

```kotlin
class Hero(val name: String) {
    fun friends(first: Int?, after: Int?): List<Hero> {
        // Your logic to call your other APIs or DB with inputs
    }
}
```

**Reply by zahari-palazov-gorillas** [reply]
Hi ExpediaGroup team, do you think it is possible to implement this specification https://relay.dev/graphql/connections.htm which is based on the https://graphql.org/learn/pagination suggested best practices by using Expedia group Spring Server library?

**Reply by smyrick** [reply]
@zahari-palazov-gorillas As mentioned above, the implementation of pagination server-side is still going to be up to you to code. We maybe could help by declaring some of the Relay models, but they would just be the models and you still have to write the logic to take the input and actually do pagination. 

Since they wouldn't have logic there isn't much benefit for us to include them in the base library. If you wanted to create your own library in Kotlin that implements the relay spec and uses graphql-kotlin we would be happy to link to it in the docs
